### PR TITLE
Upgrade to Electron Five (5)

### LINKED ISSUE #6888: upgrade app to use Electron 5
This should be a smoother upgrade than the Electron 3 process because they use the same version of Node, but #6883 is a known blocker that needs to happen too.

The other issue I'm tracking is https://github.com/electron/spectron/issues/346 which is about using Spectron with Electron 4 - there's a workaround in the issue to get it working with pre-release versions, which is unsatisfying. I'm trying to chase what the status is of this issue.

Given how subtle #6883 was, I'd recommend whoever looks at this issue should try upgrading locally to see if there's any other breakage.

### PR DESCRIPTION
**Pull request by outofambit**
## Overview

closes #6888 

## Description

- addresses breaking API changes

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -316,10 +316,10 @@
   resolved "https://registry.yarnpkg.com/@types/node/-/node-8.0.47.tgz#968e596f91acd59069054558a00708c445ca30c2"
   integrity sha512-kOwL746WVvt/9Phf6/JgX/bsGQvbrK5iUgzyfwZNcKVFcjAUVSpF9HxevLTld2SG9aywYHOILj38arDdY1r/iQ==
 
-"@types/node@^8.0.24":
-  version "8.10.40"
-  resolved "https://registry.yarnpkg.com/@types/node/-/node-8.10.40.tgz#4314888d5cd537945d73e9ce165c04cc550144a4"
-  integrity sha512-RRSjdwz63kS4u7edIwJUn8NqKLLQ6LyqF/X4+4jp38MBT3Vwetewi2N4dgJEshLbDwNgOJXNYoOwzVZUSSLhkQ==
+"@types/node@^10.12.18":
```
Comment (shiftkey) [inline yarn.lock]: Can we also bump our `@types/node` dependency in here to a 10.x version?

Comment (shiftkey) [reply yarn.lock]: Actually, let's not worry about that until we decide to move the CI builds to Node 10 or later

**Comment by outofambit** [comment]
@niik could you take a look at this and let me know if it seems sensible? i annotated most of the changes to help give context!

**Comment by outofambit** [comment]
also @tierninho if you could do a smoke test of some sort and make sure nothing obvious is broken? the plan is to have this on beta for a while, so we'll have time to find other things too. (i looked and didn't notice anything but 🤷‍♀.)

File: app/src/lib/menu-item.ts [hunk 2]
```diff
@@ -17,7 +17,43 @@ export interface IMenuItem {
    * When specified the click property will be ignored.
    * See https://electronjs.org/docs/api/menu-item#roles
    */
-  readonly role?: string
+  readonly role?:
```
[the remark this answers is not in the document]
Comment (j-f1) [reply app/src/lib/menu-item.ts]: Is there a way to pull this directly from Electron so it doesn’t get out of date? Perhaps `Electron.MenuItem['role']`?

File: app/src/lib/shell.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ async function getRawShellEnv(): Promise<string | null> {
       cleanup()
     }, 5000)
 
-    const options = {
+    const options: ChildProcess.SpawnOptions = {
```
Comment (j-f1) [inline app/src/lib/shell.ts]: Is there a reason this is pulled out into a constant rather than inlined directly into the `ChildProcess.spawn` call?

Comment (outofambit) [reply app/src/lib/shell.ts]: i don't think so. moved! that does look nicer, thanks!

**Comment by tierninho** [comment]
In the meantime, I'm gonna do a flyover on the open Electron 5.0x bugs from the [Electron repo](https://github.com/electron/electron/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+5+label%3A5-0-x+label%3A%22bug+%3Abeetle%3A%22+) to see if any might apply to us.

**Pull request merged**